swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/AbstractOperationGenerator.java

```diff
@@ -0,0 +1,594 @@
+      ParameterGenerator propertyParameterGenerator = new ParameterGenerator(this,
+          methodAnnotationMap,
+          propertyDefinition.getName(),
+          annotations.toArray(new Annotation[annotations.size()]),
+          propertyDefinition.getField().getAnnotated().getGenericType());
+      parameterGenerators.add(propertyParameterGenerator);
+    }
+  }
+
+  protected void initRemainMethodAnnotationsParameterGenerators(Map<String, List<Annotation>> methodAnnotationMap) {
+    for (Entry<String, List<Annotation>> entry : methodAnnotationMap.entrySet()) {
+      ParameterGenerator parameterGenerator = new ParameterGenerator(this, entry.getKey(), entry.getValue());
+      parameterGenerators.add(parameterGenerator);
+    }
+  }
+
+  @Override
+  public List<Annotation> collectParameterAnnotations(Annotation[] parameterAnnotations,
+      Map<String, List<Annotation>> methodAnnotationMap, String parameterName) {
+    List<Annotation> methodAnnotations = methodAnnotationMap.remove(parameterName);
+    if (methodAnnotations == null) {
+      methodAnnotations = Collections.emptyList();
+    }
+
+    List<Annotation> annotations = new ArrayList<>();
+    Collections.addAll(annotations, parameterAnnotations);
+    annotations.addAll(methodAnnotations);
+
+    return annotations;
+  }
+
+  private Map<String, List<Annotation>> initMethodAnnotationByParameterName() {
+    Map<String, List<Annotation>> methodAnnotations = new LinkedHashMap<>();
+    for (Annotation annotation : method.getAnnotations()) {
+      if (annotation instanceof ApiImplicitParams) {
+        for (ApiImplicitParam apiImplicitParam : ((ApiImplicitParams) annotation).value()) {
+          addMethodAnnotationByParameterName(methodAnnotations, apiImplicitParam.name(), apiImplicitParam);
+        }
+        continue;
+      }
+
+      if (annotation instanceof ApiParam) {
+        addMethodAnnotationByParameterName(methodAnnotations, ((ApiParam) annotation).name(), annotation);
+      }
+    }
+    return methodAnnotations;
+  }
+
+  private void addMethodAnnotationByParameterName(Map<String, List<Annotation>> methodAnnotations, String name,
+      Annotation annotation) {
+    if (StringUtils.isEmpty(name)) {
+      throw new IllegalStateException(
+          String.format("%s.name should not be empty. method=", annotation.annotationType().getSimpleName(), method));
+    }
+
+    methodAnnotations.computeIfAbsent(name, n -> new ArrayList<>())
+        .add(annotation);
+  }
+
+  @Override
+  public Type collectGenericType(List<Annotation> annotations, Type defaultType) {
+    Type genericType = null;
+    for (Annotation annotation : annotations) {
+      ParameterProcessor<Parameter, Annotation> processor = swaggerGenerator
+          .findParameterProcessors(annotation.annotationType());
+      if (processor == null) {
+        continue;
+      }
+
+      Type type = processor.getGenericType(annotation);
+      if (type != null) {
+        genericType = type;
```

Comment:
why not break when type != null, want the later one covers the former?

Reply:
yes, always override.